Metadata/CIP-Certification-Metadata.md

```diff
@@ -0,0 +1,180 @@
+
+DApp stores and light wallets will be able to pull DApps information from DApps registration or chain exploring and will link a DApp to a corresponding set of certificates.
+
+End-user will interact with a DApp through a wallet and will be able to check the different certificates obtained by the DApp.
+
+
+## Specification
+<!-- The technical specification should describe the proposed improvement in sufficient technical detail. In particular, it should provide enough information that an implementation can be performed solely on the basis of the design in the CIP. This is necessary to facilitate multiple, interoperable implementations. -->
+
+### Certification providers
+Certification providers will present which level of certification was reached and present evidence of the work done. Certification providers will sign the certificate to attest that they have done the work and prevent certificate forgeries.
+
+
+### Definitions
+- **dApp** - A decentralised application that is described by the combination of metadata, certificate and a set of used scripts.
+- **dApp Store** - A dApp aggregator application which follows on-chain data looking for and verifying dApp metadata claims, serving their users linked dApp metadata.
+
+
+### Properties
+
+**subject**, an UTF-8 encoded string, is a mandatory field which is an identifier of the claim subject.
+
+**certLevel**, an integer between 1 and 3, is a mandatory field that represents the level of certification reached in a certificate.
+
+**certificateIssuer**, a string, is a mandatory field that represents the company that has issued the certificate. It will be used with a public list of public keys to ensure that the certificate originates from a trusted certification provider. 
+
+**reportURLs**, an array of URLs, is a mandatory field that link to the actual certification report for anyone to read. This ensures transparency in the findings, what was and was not considered in the certification process.
+
+**reportHash**, a string, is a mandatory field that is the blake2b-256 hash of the audit file pointed by the links in **reportURLs**. 
+
+**summaryURLs**, an array of URLs, is a mandatory field that link to the actual certification summary for anyone to read. 
+
+**summaryHash**, a string, is a mandatory field that is the blake2b-256 hash of the summary file pointed by the links in **summaryURLs**. 
+
+**disclaimerURLs**, an array of URLs, is a mandatory field that link to the legal disclaimer about the content of the report pointed by **reportURLs** links.
+
+**disclaimerHash**, a string, is a mandatory field that is the blake2b-256 hash of the summary file pointed by the links in **disclaimerURLs**. 
```

Comment:
Are these fields really necessary? Shouldn't they be embedded in the report? 
Instead, we could have a summary that is a short string summarizing the work that has been conducted. This could be show directly on the dApp store.

Reply:
They should be embedded into the report in my opinion but there is a demand from DApp store to be able to display them directly at the DApp store level. I don't think we'll be able to standardized the format of the report to a point where they could extract it automatically which is why I made them as fields.

Reply:
In this suggestion, though, they are not embedded in the metadata which only contains a link (or several links) to the summary or the disclaimer. Is the dApp store going to pull those links, hoping one of them resolves, and show the result? Does that imply constraints on what's at the end of the link (as in plain text for instance, with a size limit)?

Reply:
I moved those to strings. Thanks for the suggestion.